packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1145,13 +1153,43 @@ export class ContainerRuntime
 		this.submitBatchFn = submitBatchFn;
 		this.submitSummaryFn = submitSummaryFn;
 		this.submitSignalFn = submitSignalFn;
+		this._getAttachState = () => context.attachState;
 
 		this.options = options;
 		this.clientDetails = clientDetails;
+		const isDetachedClient = this.clientDetails.type === detachedClientType;
+		if (isDetachedClient) {
+			this.submitFn = () => 0;
+			this.submitBatchFn = () => 0;
+			this.submitSummaryFn = () => 0;
+			this.submitSignalFn = () => {};
+		}
+
+		this.loadDetachedAndTransitionFn = async () => {
```

Comment:
This feels a bit convoluted. On one hand I can see why you might want to put this code here - "/_detachedContainerRuntime" string seems to be ContainerRuntime business only. But otherwise this block seems like does not belong here - it belongs where loadDetachedAndTransitionFn() is called.

Maybe one reason why it's not there - this class has access to loader, which (if I got right) is relative loader (i.e. it operates only on current URI namespace). But do we need loader at all? Can we have detached context and runtime attached to it, and that's about it? I.e. could we avoid touching loader layer?

Your approach with having to use loader to create container object is fine as well, it's just so hard to follow the network of connections / code flow on how things are created and what is their state as functionality is spread out.